ops/pebble.py

```diff
@@ -1363,24 +1632,27 @@ def ack_warnings(self, timestamp: datetime.datetime) -> int:
         return resp['result']
 
     def get_changes(
-        self, select: ChangeState = ChangeState.IN_PROGRESS, service: str = None,
-    ) -> typing.List[Change]:
+        self, select: ChangeState = ChangeState.IN_PROGRESS, service: Optional[str] = None,
+    ) -> List[Change]:
         """Get list of changes in given state, filter by service name if given."""
-        query = {'select': select.value}
+        query = {'select': select.value}  # type: Dict[str, Union[str, int]]
         if service is not None:
             query['for'] = service
-        resp = self._request('GET', '/v1/changes', query)
+        resp = typing.cast('_GetChangesDict', self._request('GET', '/v1/changes', query))
         return [Change.from_dict(c) for c in resp['result']]
 
     def get_change(self, change_id: ChangeID) -> Change:
         """Get single change by ID."""
-        resp = self._request('GET', '/v1/changes/{}'.format(change_id))
+        resp = typing.cast('_GetChangeDict',
+                           self._request('GET', '/v1/changes/{}'.format(change_id)))
         return Change.from_dict(resp['result'])
 
     def abort_change(self, change_id: ChangeID) -> Change:
         """Abort change with given ID."""
         body = {'action': 'abort'}
-        resp = self._request('POST', '/v1/changes/{}'.format(change_id), body=body)
+        resp = typing.cast('_GetChangeDict',
+                           self._request('POST', '/v1/changes/{}'.format(change_id),
+                                         body=body))
```

Comment:
we do need the casting here because the return object of requests is untyped (json)
so we need to clarify the assumption that the return type is some specific TypedDict.
Another option would be to do unmarshalling and validation.